### DISCUSSION TITLE
Performance of oneshot channel vs Notify when no data is needed

### DISCUSSION THREAD

**Discussion by Hawk777** [tokio-rs/tokio#5454] [Q&A]
Hi! I’m in a situation where I need to notify a task of an event, and I need to do that exactly once. Under the assumption that the “weakest” tool (the one that provides the fewest features or the most limited usage constraints) is probably the most performant or lightest-weight, there appear to be two options with no obvious choice between them: a `oneshot` channel over unit (`()`) type, or a `Notify`. It’s not obvious how to choose between them: a `oneshot` channel has the advantage of not needing the complexity of being used more than once, but a `Notify` has the advantage of not needing the complexity of transporting data. As far as I can tell, the documentation doesn’t really suggest which of these two tools is more suitable. Which is it? As this does not seem like an unusual situation, would it be appropriate to add something to the documentation about it? Thanks.

**Comment by Darksonn** [comment]
I would expect that the oneshot channel is the more performant choice.

**Reply by Hawk777** [reply]
Thanks, is that something that makes sense to make a documentation PR regarding?
